### PR TITLE
Id 206 confirm po route page donor#minor

### PR DESCRIPTION
**Pull request by spinhalf-com**
## Purpose

_Briefly describe the purpose of the change._

Fixes ###-####

## Approach

_Explain how your code addresses the purpose of the change._

## Learning

_Any tips and tricks, blog posts or tools which helped you._

## Checklist

* [x ] I have performed a self-review of my own code
* [ x] I have added relevant logging with appropriate levels to my code
* [ x] I have updated documentation where relevant
* [ x] I have added tests to prove my work
* [ x] I have run an accessibility tool against the changes and applied fixes
* [ ] The team have tested these changes

### CODE REVIEW

File: service-front/module/Application/view/application/pages/post_office/confirm_post_office.twig [hunk 1]
```diff
@@ -51,8 +51,14 @@
...
                             <dd class="govuk-summary-list__actions">
-                                <a class="govuk-link" href="post-office-documents">Change<span
-                                            class="govuk-visually-hidden"></span></a>
+                                <a
+                                    class="govuk-link"
+                                    href="post-office-documents"
+                                    aria-label="Change document"
+                                >
+                                    Change
+                                    <span class="govuk-visually-hidden"></span>
```
Comment (mattmachell) [inline service-front/module/Application/view/application/pages/post_office/confirm_post_office.twig:60]: This span should contain what the user is changing for accessibility needs where they can't see the link in context.

File: service-front/module/Application/view/application/pages/post_office/find_post_office_branch.twig [hunk 2]
```diff
@@ -29,50 +29,58 @@
...
-                            {{ item.address }} <br>
-                        </label>
+            <div class="govuk-form-group {% if form.get('postoffice').messages %}govuk-form-group--error{% endif %}">
+                {% if form.get('postoffice').messages %}
+                    <p id="dln-error" class="govuk-error-message">
+                        <span class="govuk-visually-hidden">Error:</span>
+                        {{ form.get('postoffice').messages | join(', ', ' and ') }}
+                    </p>
+                {% endif %}
+                <fieldset class="govuk-fieldset" aria-describedby="contact-hint">
```
Comment (mattmachell) [inline service-front/module/Application/view/application/pages/post_office/find_post_office_branch.twig:44]: For the `aria-describedby` you need a corresponding element with that id.

File: service-front/module/Application/src/Services/OpgApiService.php [hunk 3]
```diff
@@ -225,124 +225,23 @@ public function updateCaseWithLpa(string $uuid, string $lpa, bool $remove = fals
...
-        } else {
-            throw new OpgApiException('Post office reference code not found');
+        try {
+            $this->makeApiRequest("/counter-service/branches", 'POST', $data);
+        } catch (OpgApiException $opgApiException) {
+            throw new OpgApiException($opgApiException->getMessage());
         }
+        return $this->responseData;
+
+        // NEED TO INCLUDE PAGINATION LOGIC HERE AS WELL
```
Comment (mattmachell) [inline service-front/module/Application/src/Services/OpgApiService.php:241]: Probably don't need pagination as the max branches returned is 10.

File: service-front/module/Application/src/Forms/PostOfficeAddress.php [hunk 4]
```diff
@@ -6,15 +6,17 @@
 
 use Laminas\Form\Annotation;
 use Laminas\Hydrator\ObjectPropertyHydrator;
+use Laminas\Validator\NotEmpty;
 
 /**
  * @psalm-suppress MissingConstructor
  */
 #[Annotation\Hydrator(ObjectPropertyHydrator::class)]
-class PostOfficeNumericCode
```
Comment (mattmachell) [inline service-front/module/Application/src/Forms/PostOfficeAddress.php:14]: Switching this from the fad code to the full address, is this going to have any downstream consequences?

File: service-front/module/Application/src/Controller/DonorPostOfficeFlowController.php [hunk 5]
```diff
@@ -82,105 +82,61 @@ public function doDetailsMatchAction(): ViewModel
...
     {
+        $templates = ['default' => 'application/pages/post_office/find_post_office_branch'];
         $view = new ViewModel();
         $uuid = $this->params()->fromRoute("uuid");
 
         $optionsdata = $this->config['opg_settings']['post_office_identity_methods'];
         $detailsData = $this->opgApiService->getDetailsData($uuid);
-        $form = (new AttributeBuilder())->createForm(PostOfficeNumericCode::class);
+        $form = (new AttributeBuilder())->createForm(PostOfficeAddress::class);
         $locationForm = (new AttributeBuilder())->createForm(PostOfficeSearchLocation::class);
```
Comment (mattmachell) [inline service-front/module/Application/src/Controller/DonorPostOfficeFlowController.php:94]: General comment, but I think some of the variable naming around postcode/location/postoffice could lead to confusion as it took me a while to parse.

File: service-front/module/Application/src/Controller/DonorPostOfficeFlowController.php [hunk 6]
```diff
@@ -82,105 +82,61 @@ public function doDetailsMatchAction(): ViewModel
...
-        return $view->setTemplate('application/pages/post_office/find_post_office');
-    }
-
     public function findPostOfficeBranchAction(): ViewModel|Response
     {
+        $templates = ['default' => 'application/pages/post_office/find_post_office_branch'];
         $view = new ViewModel();
         $uuid = $this->params()->fromRoute("uuid");
 
         $optionsdata = $this->config['opg_settings']['post_office_identity_methods'];
```
Comment (mattmachell) [inline service-front/module/Application/src/Controller/DonorPostOfficeFlowController.php:91]: Do we need `optionsdata` in the postoffice lookup page? I couldn't see it used in the twig.

File: service-front/module/Application/src/Forms/PostOfficeAddress.php (hunk 4 above)
Comment (spinhalf-com) [reply service-front/module/Application/src/Forms/PostOfficeAddress.php:14]: The attrType in the API was always a string, so it should be fine (and works in testing without any issues)

**Review by mattmachell — APPROVED** [review]

**Pull request merged**